### ISSUE TITLE
Unable to run 'for /f' loop command in 'Exec' task 

### ISSUE THREAD

**Issue by honzakostejn** [dotnet/msbuild#6194]
I'm trying to run 'Exec' task without success, not sure if there's something wrong or this is not provided.

```
<Exec WorkingDirectory="$(NuGetPackageRoot)talxis.sdk.tooling.cds" ConsoleToMSBuild="true" Command="set &quot;NewestPackageVersion=&quot; &amp;&amp; for /f &quot;tokens=5 delims= &quot; %G in ('dir ^| sort /r') do if not defined NewestPackageVersion set &quot;NewestPackageVersion=%G&quot;^
      echo %NewestPackageVersion%" />
```

The command property is escaped, but when we unescape it, this is what you'll get:
```
set "NewestPackageVersion=" && for /f "tokens=5 delims= " %G in ('dir ^| sort /r') do if not defined NewestPackageVersion set "NewestPackageVersion=%G"^
      echo %NewestPackageVersion%
```
['for /f' loop](https://ss64.com/nt/for_r.html)

If I run this in my cmd from the same path, I'll get the result I need:
![image](https://user-images.githubusercontent.com/32060653/109002298-51906b80-76a6-11eb-832c-733254b3a853.png)

If I run the command in exec as a part of build target, I'm getting this error:
`G"^ was unexpected at this time`

Okay, let's say we won't use the parameter `%G` for the time being and replace it with some test value:
```
set "NewestPackageVersion=" && for /f "tokens=5 delims= " %G in ('dir ^| sort /r') do if not defined NewestPackageVersion set "NewestPackageVersion=TEST_VALUE"^
      echo %NewestPackageVersion%
```

This acts as expected:
![image](https://user-images.githubusercontent.com/32060653/109002865-0e82c800-76a7-11eb-8477-eb80a31f3876.png)

So now my 'Exec' looks like this:
```
<Exec WorkingDirectory="$(NuGetPackageRoot)talxis.sdk.tooling.cds" ConsoleToMSBuild="true" Command="set &quot;NewestPackageVersion=&quot; &amp;&amp; for /f &quot;tokens=5 delims= &quot; %G in ('dir ^| sort /r') do if not defined NewestPackageVersion set &quot;NewestPackageVersion=TEST_VALUE&quot;^
      echo %NewestPackageVersion%" />
```

If I run the build target, this is what I get:
`G was unexpected at this time.`
So it seems like you can't use the for loop parameter, I've also tried replacing `%G` with `%%G` like in scripts. When I change the letter to `X`, I get `The system cannot find the path specified.` from the last echo, which confirms that there's something happening sooner.

Am I doing something wrong, or this is not supported? I'm switching to PowerShell inside the exec command, since that will work for me right now, but I'm still curious. Thanks.

**Comment by Forgind** [comment]
I think you have to replace `%` with `&percnt;`.

**Comment by honzakostejn** [comment]
![image](https://user-images.githubusercontent.com/32060653/109635283-7a06e280-7b4a-11eb-99a6-c34590959925.png)

no luck here. I even tried using "%25"
https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-special-characters?view=vs-2019

**Comment by Forgind** [comment]
With your repro, I was able to get it to fail how you said it would, and when I double percent signs indicating variables, it stopped failing. I don't know if it was doing what was expected, but maybe? So:
`<Exec WorkingDirectory="$(NuGetPackageRoot)talxis.sdk.tooling.cds" ConsoleToMSBuild="true" Command="set &quot;NewestPackageVersion=&quot; &amp;&amp; for /f &quot;tokens=5 delims= &quot; %%G in ('dir ^| sort /r') do if not defined NewestPackageVersion set &quot;NewestPackageVersion=%%G&quot;^
      echo %NewestPackageVersion%" />`

Also not entirely sure why that made it work, but I noticed that if I paused mid-execution and copied what Exec had written into the batch file into a command prompt, it failed. I'm guessing it needed two levels of escaping, and that was the second level.
